Patient Image

yujuncho opened the pull request:
Created components to pick and upload patient images in /main/history and /main/newVisit. API routes and multer package were also added to support this functionality. This PR also makes some adjustments to the responsiveness of /main/history and /main/newVisit.

Note, images are currently saved in a local directory on the server.

yujuncho committed:
Added image picker to /main/history and /main/newVisit

yujuncho committed:
- Added route for uploading profile images for patients
- Updated the client to upload patient images accordingly

yujuncho committed:
Updated responsiveness of /main/history and /main/newVisit

yujuncho committed:
Changed updatePatientImage to delete the old image

yujuncho committed:
Added an uploads/images directory in the repo, so image uploads are
temporarily saved on Heroku

yujuncho committed:
Fixed a bug where updating a patient's profile image did not update
useHistory's state, so on a page refresh, the patient profile image did
not properly use the updated image. Also the height of the profile image
for more consistent responsive behavior.

adbahrani commented on client/src/ehr_components/PatientImageUpload.tsx:
```diff
@@ -0,0 +1,43 @@
+
+      history.replace(pathname, { ...patientState, profileImage });
+    } catch (error) {
+      console.log(error);
```
You might wanna show error to user as well

adbahrani commented on client/src/ehr_components/PatientImageUpload.tsx:
```diff
@@ -0,0 +1,43 @@
+        data: {
+          patient: { profileImage }
+        }
+      } = await Axios.patch("/api/patient/image", formData);
```
Nice use of path here

yujuncho committed:
Changed file limit for patient image uploads to MongoDB's size limit, and created patientImage model, routes, and controller

yujuncho committed:
- Fixed bugs with api calls to uploading a patient image
- Created a GET route for retrieving patient images
- Updated client-side functionality to load existing patient images and
to upload new ones

yujuncho committed:
- Brought back history.replace for PatientImageUpload for the reload bug
fix. (Reload would use the old patient state).
- Added the toastr to PatientImageUpload to display errors
- Customized multer error handling for a more specific message when the
file is too large
